projects/com.oracle.truffle.llvm.parser.bc/src/com/oracle/truffle/llvm/parser/bc/LLVMBitcodeVisitor.java

```diff
@@ -348,10 +359,74 @@ private LLVMExpressionNode getGlobalVariable(GlobalValueSymbol global) {
             final LLVMExpressionNode loadedFunction = factoryFacade.createLoad(this, functionType, functionLoadTarget);
             final LLVMExpressionNode[] argNodes = new LLVMExpressionNode[]{factoryFacade.createFrameRead(this, LLVMBaseType.ADDRESS, stackSlot)};
             final LLVMNode functionCall = factoryFacade.createFunctionCall(this, loadedFunction, argNodes, LLVMBaseType.VOID);
-            structors[i] = functionCall;
+
+            final StructureConstant structureConstant = (StructureConstant) arrayConstant.getElement(i);
+            final Integer priorityConstant = LLVMSymbolResolver.evaluateIntegerConstant(structureConstant.getElement(0));
+            assert priorityConstant != null;
+            sortedStructors.add(priorityConstant, functionCall);
+        }
+
+        return sortedStructors.getAll();
+    }
+
+    private static final class SortedLLVMNodeList {
```

Comment:
Java's PriorityQueue does not support priorities separate from the data objects and the specification of [toArray()](https://docs.oracle.com/javase/8/docs/api/java/util/PriorityQueue.html#toArray--) explicitly states that the elements do not follow a particular order. Considering these drawbacks I decided that an explicit implementation would require the least code.